### ISSUE TITLE
Avoid creation of very big changesets

### ISSUE THREAD

**Issue by Mahabarata** [openstreetmap/iD#9647]
### Description

Hi,

I don't know if it is easy to do : I just made a big mistake, I improved some part of map in west US, forgot to save my changes and changed something else (just a node) in Ireland then saved it. It created of course a huge area for my changeset. I let a comment on it to apologize.

When we want to save a changeset, we have warnings about our changes (if we forgot a role in a multipolygon for example) : could you add a warning when the area of a changeset is very big ?

Of course "big" is not a very clear word, but a warning is just a warning : if I improve a river like the Nile, the changeset could be big. But to have a warning will not prevent the save. So better to have one than nothing.

The first step would be to only have a warning when the area is big (I suggest : when a side of the changeset is bigger than 100 km. Just my idea, choose what you want).

The second step would be to tell us why ("this element is very far from others") : more difficult probably, so maybe it can wait.

Or the second step would be to be able to see the area of the changeset with our changes like when we use the G option.

Best regards

**Comment by jacobwhall** [comment]
I really like this idea. It happens to me sometimes too, and I feel bad lumping a bunch of random geographically-dispersed changes together into one changeset. A warning dialog (perhaps with a "ignore this warning in the future" checkbox) would be helpful to me.

**Comment by Dimitar5555** [comment]
> The first step would be to only have a warning when the area is big (I suggest : when a side of the changeset is bigger than 100 km. Just my idea, choose what you want).

IMHO the first and last step should be to not allow editing that far away in the first place. It should be possible to create a square box around the current area with a side of 100 km. If the user tries to go out of it, make the map darker/place a black overlay over it and show a message "The area you are trying to edit now is very far from where you started. Upload your current changes now and you will be able to edit here." or something similar. Such thing is already implemented in RapiD, but the blocked area is not dynamic.

**Comment by Mahabarata** [comment]
Yes, good idea, better than mine : like with vaccines, better to prevent than to cure.

**Comment by HikeAndMap** [comment]
I'm not going into details how to make this work - but I support the proposal generally that something must be implemented to limit the boxes

**Comment by dreua** [comment]
This would benefit two user groups: New users who don't know better and also those who want to do QA and subscribe to changes covering a certain area. Mistakes like this seem to happen about once per week, two happened today:

https://www.openstreetmap.org/changeset/148342592#map=7/51.918/10.547
https://www.openstreetmap.org/changeset/148349978

I think a warning or block when editing is best, this allows users to save their old changes and continue with what they wanted to do. For new users (very few changesets) there could be a hard block imo, but that can also be discussed after a warning is implemented.

**Comment by dreua** [comment]
Maybe the title of this issue could be improved too: "Avoid creation of very big changesets" or something. Currently it lacks context, it could also be a title for a performance issue with big changesets or something.

**Comment by Mahabarata** [comment]
I won't open a new issue but I see now that this problem of big changesets is more general than an iD issue and has a very bad consequence : the "History" button (at the top of the map) is, almost always, useless !

Go anywhere on the map (choose a fairly small area for example https://www.openstreetmap.org/edit#map=14/46.4972/2.7548), and click on the History button : this button should show you the lastest changes in this area.

But look at each changeset of the history : there are so many "big changesets" covering the zone, with most not a single change in this specific area but outside, that it is impossible to know what latest changes wete made in this specific area. And all these changesets are created by so many different softwares (sometimes iD editor but not the more common) that to fix this issue will of course improve iD but won't solve the whole problem.

**Comment by dreua** [comment]
@Mahabarata while there are more issues in this context I think this issue to address it during changeset creation in iD is very valid in itself. Solving it would solve a significant chunk of the whole "big changeset" issues, not all of it.

~~Since you're here, maybe you could improve this issues title per my suggestion above?~~ Edit: Done, thanks!

Off topic: [Whodidit](https://simon04.dev.openstreetmap.org/whodidit/?zoom=14&lat=50.17539&lon=-122.47764&layers=BTT&age=1000) is a tool to list the latest changes in relatively small tiles, maybe that is better than the history button for your use case. Maybe the history button's result should be improved too, but that is leading us further off topic and should not be discussed in this issue on iD. [Feel free to hide this whole comment as off topic to clean up the issue.]

**Comment by Mahabarata** [comment]
Thanks for the link to whodidit.

**Comment by tyrasd** [comment]
This is a duplicate of #8590. Also since a few months, the OSM api does not allow new users to upload changeset with large bounding boxes (see #10284), so this should be a slightly less pronounced issue nowadays.
